### PR TITLE
Add test case for predicates

### PR DESCRIPTION
**Pull request by albertZhangTJ**

### CODE REVIEW

File: src/test/java/seedu/boba/model/customer/NameSoundsSimilarToPredicateTest.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        assertTrue(predicate.test(new CustomerBuilder().withName("Alice Bob").build()));
+    }
+
+    @Test
+    public void test_nameSoundsSimilarTo_returnsTrue() {
+        NameSoundsSimilarToPredicate predicate = new NameSoundsSimilarToPredicate(Collections.singletonList("Alice"));
+        assertTrue(predicate.test(new CustomerBuilder().withName("Alise").build()));
+
+        predicate = new NameSoundsSimilarToPredicate(Arrays.asList("Aschcroft"));
+        assertTrue(predicate.test(new CustomerBuilder().withName("Ashcraft").build()));
```
Comment (WuHaohui1231) [inline src/test/java/seedu/boba/model/customer/NameSoundsSimilarToPredicateTest.java:66]: Maybe can add some names that are more common and easy to mix up, like Steven and Stephen

File: src/test/java/seedu/boba/model/customer/AllInfoContainsKeywordsTest.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+        // null -> returns false
+        assertFalse(firstPredicate.equals(null));
+
+        // different customer -> returns false
+        assertFalse(firstPredicate.equals(secondPredicate));
+    }
+
+    @Test
+    public void test_nameContainsKeywords_returnsTrue() {
```
Comment (TanWenCong) [inline src/test/java/seedu/boba/model/customer/AllInfoContainsKeywordsTest.java:45]: I like how you used comments to partition the possible cases

**Review by TanWenCong — APPROVED** [review]
LGTM! :^)

**Review by WuHaohui1231 — APPROVED** [review]
LGTM!

File: src/test/java/seedu/boba/model/customer/AllInfoContainsKeywordsTest.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        assertFalse(predicate.test(new CustomerBuilder().withName("Alice").build()));
+
+        // Non-matching keyword
+        predicate = new AllInfoContainsKeywordsPredicate(Arrays.asList("Carol"));
+        assertFalse(predicate.test(new CustomerBuilder().withName("Alice Bob").build()));
+
+        // Keywords match phone, email and address, but does not match name
+        predicate = new AllInfoContainsKeywordsPredicate(Arrays.asList("12345678", "alice@email.com", "420"));
+        assertTrue(predicate.test(new CustomerBuilder().withName("Alice").withPhone("12345678")
+                .withEmail("alice@email.com").withReward("420").build()));
```
Comment (WuHaohui1231) [inline src/test/java/seedu/boba/model/customer/AllInfoContainsKeywordsTest.java:77]: Nice coverage improvement!

File: src/test/java/seedu/boba/model/customer/NameSoundsSimilarToPredicateTest.java (hunk 1 above)
Comment (albertZhangTJ) [reply src/test/java/seedu/boba/model/customer/NameSoundsSimilarToPredicateTest.java:66]: Ok!

**Pull request merged**